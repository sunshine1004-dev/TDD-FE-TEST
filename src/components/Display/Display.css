.display-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 0 4%;
}

.display-value {
  font-family: var(--main-font);
  margin-left: auto;
  overflow: hidden;
}
