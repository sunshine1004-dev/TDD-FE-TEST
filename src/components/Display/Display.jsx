import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ({ sum, average, deviation }) => (
    <div className="display-container">
        <p className="display-value">sum: {sum}</p>
        <p className="display-value">average: {average}</p>
        <p className="display-value">deviation: {deviation}</p>
    </div>
);

Display.propTypes = {
    sum: PropTypes.string.isRequired,
    average: PropTypes.string.isRequired,
    deviation: PropTypes.string.isRequired,
};

Display.defaultProps = { sum: '0', average: '0', deviation: '0' };

export default Display;
