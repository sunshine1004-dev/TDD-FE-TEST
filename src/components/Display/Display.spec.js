import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

const MOCK_DISPLAY_VALUE = 'a display value';

describe('Display', () => {
  let wrapper;
  beforeEach(
    () =>
      (wrapper = shallow(
        <Display
          sum={MOCK_DISPLAY_VALUE}
          average={MOCK_DISPLAY_VALUE}
          deviation={MOCK_DISPLAY_VALUE}
        />
      ))
  );

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
});
