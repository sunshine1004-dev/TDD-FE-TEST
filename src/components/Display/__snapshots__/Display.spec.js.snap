// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render correctly 1`] = `
<div
  className="display-container"
>
  <p
    className="display-value"
  >
    sum: 
    a display value
  </p>
  <p
    className="display-value"
  >
    average: 
    a display value
  </p>
  <p
    className="display-value"
  >
    deviation: 
    a display value
  </p>
</div>
`;
